packages/inter-protocol/src/clientSupport.js

```diff
@@ -232,11 +232,16 @@ const makeBidOffer = (brands, opts) => {
   // @ts-expect-error XXX how to narrow AssetKind?
   const collateralBrand = brands[opts.collateralBrandKey];
 
-  const want = AmountMath.make(
+  const wantAmt = AmountMath.make(
     collateralBrand,
     scaleDecimals(opts.wantCollateral),
   );
 
+  const proposal = {
+    give,
+    ...('price' in opts ? { want: { Collateral: wantAmt } } : {}),
```

Comment:
I think `want` shouldn't be added to the proposal by default. It turns the bid into an all-or-nothing offer. If the auction tries to partially satisfy the offer, zoe will abort the transaction as not-offer-safe.

Reply:
Help me understand how the contract works in this respect?

Do you have a suggestion on what the CLI should look like?

Reply:
Is there something in the contract that suggested the use of `want` for this in the CLI?  I thought the contract's terminology around this was `bidScaling` and `price`.

Reply:
> Is there something in the contract that suggested the use of `want` for this in the CLI? I thought the contract's terminology around this was `bidScaling` and `price`.

https://github.com/Agoric/agoric-sdk/blob/b834e89d39edf2af1a0a730f33b68494fe0c9ee2/packages/inter-protocol/src/auction/auctioneer.js#L275

Reply:
oops... rather:

https://github.com/Agoric/agoric-sdk/blob/b834e89d39edf2af1a0a730f33b68494fe0c9ee2/packages/inter-protocol/src/auction/auctionBook.js#L71

Reply:
Sorry, Dan. You're right. The contract specifies that. (oops)

The replacement for `want` in the offerArgs would be something more like `targetPurchase` or `desiredBuy`.  I agree that `want` is bad here. What should we change it to?

Reply:
`desiredBuy` is closer to the style of `give`/`want` while being distinct enough. also "desired" makes clear that it's not a guarantee the way `want` is.